### PR TITLE
Preserve all Book and Label data in JSON files -

### PR DESCRIPTION
**Pull request by anshuman7negi**
**In this branch I have added the following changes:**

- Added a method to save data into a .json file.
- Added a method to load data from the .json file.
- All data has been preserved by saving collections in .json files.

### CODE REVIEW

File: data/preserve_data.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require 'json'
+
+def save_book(data, filename)
+  data_array = data.map(&:to_h)
+  File.write(filename, JSON.generate(data_array))
+end
+
+def save_label(_data, filename)
+  data_array = ddata.map(&:to_h)
```
Comment (balatstar) [inline data/preserve_data.rb:9]: I'm getting an error when trying to create a new label. Do you think ddata.map should be _data.map here?

File: modules/create_label.rb [hunk 2]
```diff
@@ -9,7 +9,37 @@ def create_label
...
     puts "\n"
   end
+
+  def label_info(book)
+    if @labels.empty?
+      puts 'No labels are available. Please create a label first.'
+      create_label
+    else
+      display_labels
+      puts '\nChoose a label for this book or enter "new" to add a new label:'
```
Comment (balatstar) [inline modules/create_label.rb:23]: Please remove \n from this line. It is displaying on the frontend UI.

**Review by balatstar — CHANGES_REQUESTED** [review]
Good job! We're nearly there... just a few minor issues for fixing, please see notes below. :)

File: modules/create_label.rb (hunk 2 above)
Comment (anshuman7negi) [reply modules/create_label.rb]: done

File: data/preserve_data.rb (hunk 1 above)
Comment (anshuman7negi) [reply data/preserve_data.rb]: done

**Review by JoyTaribagshaw — APPROVED** [review]
Great job implementing the changes, you can now merge this pr.🎉

**Pull request merged**